components/google-cloud/google_cloud_pipeline_components/experimental/custom_job/custom_job.py

```diff
@@ -0,0 +1,227 @@
+                component_op.component_spec.implementation.container.command,
+                _is_output_parameter
+            )
+            worker_pool_spec['container_spec'][
+                'command'
+            ] = component_op.component_spec.implementation.container.command
+
+        if component_op.component_spec.implementation.container.args:
+            dsl_utils.resolve_cmd_lines(
+                component_op.component_spec.implementation.container.args,
+                _is_output_parameter
+            )
+            worker_pool_spec['container_spec'][
+                'args'
+            ] = component_op.component_spec.implementation.container.args
+        if accelerator_type is not None:
+            worker_pool_spec['machine_spec']['accelerator_type'
+                                            ] = accelerator_type
+        if accelerator_count is not None:
+            worker_pool_spec['machine_spec']['accelerator_count'
+                                            ] = accelerator_count
+        if boot_disk_type is not None:
+            if 'diskSpec' not in worker_pool_spec:
+                worker_pool_spec['diskSpec'] = {}
+            worker_pool_spec['diskSpec']['bootDiskType'] = boot_disk_type
+        if boot_disk_size_gb is not None:
+            if 'diskSpec' not in worker_pool_spec:
+                worker_pool_spec['diskSpec'] = {}
+            worker_pool_spec['diskSpec']['bootDiskSizeGb'] = boot_disk_size_gb
+
+        job_spec['worker_pool_specs'] = [worker_pool_spec]
+        if replica_count is not None and replica_count > 1:
+            additional_worker_pool_spec = copy.deepcopy(worker_pool_spec)
+            additional_worker_pool_spec['replica_count'] = str(
+                replica_count - 1
+            )
+            job_spec['worker_pool_specs'].append(additional_worker_pool_spec)
+
+    if timeout is not None:
+        if 'scheduling' not in job_spec:
+            job_spec['scheduling'] = {}
+        job_spec['scheduling']['timeout'] = timeout
+    if restart_job_on_worker_restart is not None:
+        if 'scheduling' not in job_spec:
+            job_spec['scheduling'] = {}
+        job_spec['scheduling']['restartJobOnWorkerRestart'
+                              ] = restart_job_on_worker_restart
+    if service_account is not None:
+        job_spec['serviceAccount'] = service_account
+    if network is not None:
+        job_spec['network'] = network
+
+    custom_job_payload = {
+        'display_name': display_name or component_op.component_spec.name,
+        'job_spec': job_spec
+    }
+
+    custom_job_component_spec = ComponentSpec(
+        name=component_op.component_spec.name,
+        inputs=component_op.component_spec.inputs + [
+            InputSpec(name='gcp_project', type='String'),
+            InputSpec(name='gcp_region', type='String')
+        ],
+        outputs=component_op.component_spec.outputs.extend([
+            OutputSpec(name='gcp_resources', type='gcp_resources')
```

Comment:
Should type here be `GCPResources` ? Also, should this be added as an `Artifact` type in the SDK? Otherwise it shows up in UI as `system.Artifact`.

/cc @IronPan

Reply:
> Should type here be GCPResources 

I'm following up with @IronPan to clarify. I will update accordingly, leaving this comment open for now.  

>should this be added as an Artifact type in the SDK

Good call, I added a tracker for updating the SDK.

Reply:
Confirmed with @IronPan  the is defined to be "GCP_RESCOURCES". I have updated the code accordingly though we can change this in the future as needed.